### PR TITLE
Implement fchmod, rename(at) sandboxing for compilers

### PR DESCRIPTION
**Pull request by quantum5**

### CODE REVIEW

File: dmoj/executors/compiled_executor.py [hunk 1]
```diff
@@ -199,6 +197,42 @@ def on_return():
 
         return self.check_file_access_at('utimensat')(debugger)
 
+    def do_fchmod(self, debugger: Debugger) -> bool:
+        path = self._getfd_pid(debugger.tid, debugger.uarg0)
+        return True if self.write_fs_jail.check(path) else ACCESS_EPERM(debugger)
+
+    def do_rename(self, debugger: Debugger) -> bool:
+        old_path, old_error = self.read_path('rename', debugger, debugger.uarg0)
```
Comment (Xyene) [inline dmoj/executors/compiled_executor.py:205]: Nit: `old_path_error`? `old_error` and `new_error` sound weird.

Comment (quantum5) [reply dmoj/executors/compiled_executor.py]: Fixed.

**Review by Xyene — APPROVED** [review]

**Comment by quantum5** [comment]
retest this please

**Comment by quantum5** [comment]
retest this please

**Pull request merged**